src/iterators.rs

```diff
@@ -641,3 +641,157 @@ pub fn new_axis_iter_mut<A, D>(v: ArrayViewMut<A, D>,
+    last_dim.slice_mut()[axis] = if rem == 0 { size } else { rem };
+
+    let last_ptr = unsafe {
+        v.ptr.offset(stride * last_index as isize)
+    };
+    let iter = OuterIterCore {
+        index: 0,
+        len: shape,
+        stride: stride,
+        inner_dim: inner_dim,
+        inner_strides: v.strides.clone(),
+        ptr: v.ptr,
+    };
+
+    (iter, last_ptr, last_dim)
+}
+
+pub fn new_chunk_iter<A, D>(v: ArrayView<A, D>,
+                            axis: usize,
+                            size: usize,
+                            ) -> ChunkIter<A, D>
+    where D: Dimension
+{
+    let (iter, last_ptr, last_dim) = chunk_iter_parts(v.view(), axis, size);
+
+    ChunkIter {
+        iter: iter,
+        last_ptr: last_ptr,
+        last_dim: last_dim,
+        life: PhantomData,
+    }
+}
+
+macro_rules! chunk_iter_impl {
+    ($iter:ident, $array:ident) => (
+        impl<'a, A, D> $iter<'a, A, D>
+            where D: Dimension
+        {
+            fn get_subview(&self,
+                           iter_item: Option<*mut A>
+                          ) -> Option<$array<'a, A, D>>
+            {
+                iter_item.map(|ptr| {
+                    if ptr != self.last_ptr {
+                        unsafe {
+                            $array::new_(ptr,
+                                         self.iter.inner_dim.clone(),
+                                         self.iter.inner_strides.clone())
+                        }
+                    }
+                    else {
+                        unsafe {
+                            $array::new_(ptr,
+                                         self.last_dim.clone(),
+                                         self.iter.inner_strides.clone())
+                        }
+                    }
+                })
+            }
+        }
+
+        impl<'a, A, D> Iterator for $iter<'a, A, D>
+            where D: Dimension,
+        {
+            type Item = $array<'a, A, D>;
+
+            fn next(&mut self) -> Option<Self::Item> {
+                let res = self.iter.next();
+                self.get_subview(res)
+            }
+
+            fn size_hint(&self) -> (usize, Option<usize>) {
+                self.iter.size_hint()
+            }
+        }
+
+        impl<'a, A, D> DoubleEndedIterator for $iter<'a, A, D>
+            where D: Dimension,
+        {
+            fn next_back(&mut self) -> Option<Self::Item> {
+                let res = self.iter.next_back();
+                self.get_subview(res)
+            }
+        }
+
+        impl<'a, A, D> ExactSizeIterator for $iter<'a, A, D>
+            where D: Dimension,
+        { }
+    )
+}
+
+/// An iterator that traverses over the specified axis
+/// and yields mutable subviews of the specified size on this axis.
+///
+/// For example, in a 2 × 8 × 3 array, if the axis of iteration
+/// is 1 and the chunk size is 2, the yielded elements
+/// are 2 × 2 × 3 subviews (and there are 4 in total).
```

Comment:
we can call them views instead of "subviews" since they have the same dimensionality as the original array.